### PR TITLE
`savefig` features

### PR DESCRIPTION
**Pull request by zmoon**
* `pydecorate` is (currently) a non-optional dep, so we don't need to `try`-`import` it
* document `loc` and add the other corner as an option
* allow specifying other logo
* allow specifying logo height in order to scale it in the image
  - but to ensure quality, probably don't want to be using this too much, instead create other versions of logo
* some input validation

asdf
*Example with custom logo and logo height and `loc=3`*

### CODE REVIEW

**Comment by zmoon** [comment]
@rschwant does this cover everything you wanted?

File: monet/plots/__init__.py [hunk 1]
```diff
@@ -37,58 +37,80 @@ def _dynamic_fig_size(obj):
     return figsize
 
 
-def savefig(fname, loc=1, decorate=True, **kwargs):
-    """save figure and add the MONET logo .
+def savefig(fname, *, loc=1, decorate=True, logo=None, logo_height=None, **kwargs):
+    """Save figure and add the MONET logo.
```
Comment (zmoon) [inline monet/plots/__init__.py]:
```suggestion
    """Save figure and add logo.
```

File: monet/plots/__init__.py [hunk 2]
```diff
@@ -37,58 +37,80 @@ def _dynamic_fig_size(obj):
...
+        Must include desired file extension (``.jpg`` or ``.png``).
     loc : int
-        the location for the monet logo.
+        The location for the logo.
+
+        * 1 -- bottom left (default)
+        * 2 -- bottom right
+        * 3 -- top right
+        * 4 -- top left
     decorate : bool
```
Comment (zmoon) [inline monet/plots/__init__.py]:
```suggestion
    decorate : bool, default: True
```

File: monet/plots/__init__.py [hunk 3]
```diff
@@ -37,58 +37,80 @@ def _dynamic_fig_size(obj):
...
-            img.save(fname, "PNG")
-        elif fname.split(".")[-1] == "jpg":
-            img.save(fname, "JPEG")
+        elif loc == 4:
+            pass
+        else:
+            raise ValueError(f"invalid `loc` {loc!r}")
+        dc.add_logo(logo, **add_logo_kwargs)
+
+        # PIL will determine format from the filename extension
```
Comment (zmoon) [inline monet/plots/__init__.py]:
```suggestion
        # PIL.Image will determine format from the filename extension
```

**Comment by rschwant** [comment]
Yes, this covers everything, thanks!

**Pull request merged**